org.eclipse.buildship.gradleprop.languageserver/src/main/java/server/completion/PropertiesMatcher.java

```diff
@@ -0,0 +1,77 @@
+package server.completion;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.Position;
+
+public class PropertiesMatcher {
+
+  public static List<CompletionItem> getCompletions(String content, Position position) {
+    var completionWord = getCompletionWord(content, position);
```

Comment:
The Buildship project is compiled against a range of Eclipse versions, some of which need Java 8 bytecode. We have two options here:
- change the language level to 8 for this feature
- exclude the feature from update sites for older Eclipse targets.
I'd vote for the first option